File: todo/taskly_backend/model/todo_model.js
const mongoose=require('mongoose')
const UserModel = require('./user_model')
const { Schema } = mongoose

const TodoSchema = new mongoose.Schema({
    userId:{
        type:Schema.Types.ObjectId,
        ref:UserModel.modelName,
        required:true
    },

    category:{
        type:String,
        enum:['Educational','Grocery','Work-Related','Home-Related','Personal'],
        default:'Personal',
        lower:true
    },

    title:{
        type:String,
        required:true
    },

    description:{
        type:String,
        required:true
    },

    notes: {
        type:String,
    },

    status:{
        type:String,
        enum:['pending','completed'],
        default:'pending'
    }
});

const TodoModel=mongoose.model('TodoTasks',TodoSchema);
module.exports=TodoModel;

File: todo/lib/providers/fetchtodo/todomodel.dart

class TodoModel{
  final String id;
  final String userId;
  final String category;
  final String title;
  final String description;
  final String notes;
  final String status;
  TodoModel({required this.id,required this.userId,required this.category,required this.title,required this.description,required this.notes,required this.status});

  factory TodoModel.fromJson(Map<String,dynamic> json){
    return TodoModel(
        id: json['_id'],
        userId: json['userId'],
        category: json['category'],
        title: json['title'],
        description: json['description'],
        notes: json['notes'],
        status: json['status']
    );
  }

    Map<String,dynamic> toJson(){
      return {
        'id': id,
        'userId':userId,
        'category':category,
        'title': title,
        'description': description,
        'notes':notes,
        'status': status
      };
  }
}